state-transition/core/state_processor_sidecar_verifier.go

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: BUSL-1.1
+//
+// Copyright (C) 2024, Berachain Foundation. All rights reserved.
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file of this repository and at www.mariadb.com/bsl11.
+//
+// ANY USE OF THE LICENSED WORK IN VIOLATION OF THIS LICENSE WILL AUTOMATICALLY
+// TERMINATE YOUR RIGHTS UNDER THIS LICENSE FOR THE CURRENT AND ALL OTHER
+// VERSIONS OF THE LICENSED WORK.
+//
+// THIS LICENSE DOES NOT GRANT YOU ANY RIGHT IN ANY TRADEMARK OR LOGO OF
+// LICENSOR OR ITS AFFILIATES (PROVIDED THAT YOU MAY USE A TRADEMARK OR LOGO OF
+// LICENSOR AS EXPRESSLY REQUIRED BY THIS LICENSE).
+//
+// TO THE EXTENT PERMITTED BY APPLICABLE LAW, THE LICENSED WORK IS PROVIDED ON
+// AN “AS IS” BASIS. LICENSOR HEREBY DISCLAIMS ALL WARRANTIES AND CONDITIONS,
+// EXPRESS OR IMPLIED, INCLUDING (WITHOUT LIMITATION) WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, NON-INFRINGEMENT, AND
+// TITLE.
+
+package core
+
+import (
+	ctypes "github.com/berachain/beacon-kit/consensus-types/types"
+	"github.com/berachain/beacon-kit/primitives/common"
+	"github.com/berachain/beacon-kit/primitives/crypto"
+	"github.com/berachain/beacon-kit/primitives/version"
+)
+
+func (sp *StateProcessor[
+	_, _, BeaconStateT, _, _, _, _,
+	_, ForkDataT, _, _, _, _, _,
+]) GetSidecarVerifierFn(
```

Comment:
q: I am not sure I understand why we need this function to depend on StateProcessor.
AFAIU, we need 'BeaconState', `ChainSpec` and `signed` in order to verify a sidecar, all elements that we could pass directly to the sidecar verifier.
Can we simplify the implementation, especially after we ditched the pub/sub system?

Reply:
The BeaconState exposed to the blockchain service is not the full one you'd access to through the state processor of the same service. The former implements the `ReadOnlyBeaconState` which at the moment lacks methods needed by the verifier, e.g. `GetGenesisValidatorsRoot`, etc. I understand we could always extend the interface but I think that would miss the point.